### PR TITLE
feat: add support for getting DORA metrics from projects and groups

### PR DESCRIPTION
**Pull request by 2785**
resolves https://github.com/xanzy/go-gitlab/issues/1557

### CODE REVIEW

**Review by jimmy0012 — CHANGES_REQUESTED** [review]
Was testing this and ran into an issue. Sometimes Dora metrics are not integers so these functions will not be able to marshal the json:
`   error="error fetching DORA Metric leadTime for <repo>: json: cannot unmarshal number 607.5 into Go struct field DORAMetric.value of type int" project-name=<repo>`

**Comment by 2785** [comment]
:neutral_face:  I so wish there's a day that gitlab docs actually tell you that

File: dora_metrics.go [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+func (s *DORAMetricsService) GetProjectDORAMetrics(pid interface{}, opt GetDORAMetricsOptions, options ...RequestOptionFunc) ([]DORAMetric, *Response, error) {
+	project, err := parseID(pid)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	if opt.Metric == "" {
+		return nil, nil, fmt.Errorf("metric cannot be empty")
+	}
+
```
Comment (svanharmelen) [inline dora_metrics.go:69]:
```suggestion
```

While I understand this, I prefer to let the API return this info instead.

File: dora_metrics.go [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+	if err != nil {
+		return nil, resp, err
+	}
+
+	return metrics, resp, err
+}
+
+// GetGroupDORAMetrics gets the DORA metrics for a group.
+//
+// GitLab API Docs: https://docs.gitlab.com/ee/api/dora/metrics.html#get-group-level-dora-metrics
```
Comment (svanharmelen) [inline dora_metrics.go:88]:
```suggestion
// GitLab API Docs:
// https://docs.gitlab.com/ee/api/dora/metrics.html#get-group-level-dora-metrics
```

File: dora_metrics.go [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+func (s *DORAMetricsService) GetGroupDORAMetrics(gid interface{}, opt GetDORAMetricsOptions, options ...RequestOptionFunc) ([]DORAMetric, *Response, error) {
+	group, err := parseID(gid)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	if opt.Metric == "" {
+		return nil, nil, fmt.Errorf("metric cannot be empty")
+	}
+
```
Comment (svanharmelen) [inline dora_metrics.go:98]:
```suggestion
```

**Review by svanharmelen — CHANGES_REQUESTED** [review]

**Comment by heidiberry** [comment]
This PR can be closed without merging, the functionality was added under https://github.com/xanzy/go-gitlab/pull/1931

**Comment by svanharmelen** [comment]
Thanks for the ping @heidiberry 👍🏻

**Pull request closed without merging**